nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -101,6 +110,13 @@ public static void checkDocumentEventContext(Event event, DocumentModel expected
     public static void checkReceivedMail(List<MailMessage> mails, DocumentModel commentDocModel,
             DocumentModel commentedDocModel, Event event, String commentEventType) {
         int expectedMailsCount = COMMENT_REMOVED.equals(commentEventType) ? 0 : 1;
+
+        DocumentModel sourceDocument = ((DocumentEventContext) event.getContext()).getSourceDocument();
+
+        expectedMailsCount = COMMENT_ADDED.equals(commentEventType)
+                && !commentedDocModel.getId().equals(sourceDocument.getId()) //
+                        ? 2
+                        : expectedMailsCount;
```

Comment:
I thought it was removed?

Reply:
We worked on adding mail count check back so we needed to put it back to make sure it is solved.
I changed the ternary expression to a simple in the upcomming push